botocore/httpchecksum.py

```diff
@@ -259,7 +258,18 @@ def resolve_request_checksum_algorithm(
     params,
     supported_algorithms=None,
 ):
+    # If the header is already set by the customer, skip calculation
+    if _has_checksum_header(request):
+        return
+
+    request_checksum_calculation = request["context"][
+        "client_config"
+    ].request_checksum_calculation
     http_checksum = operation_model.http_checksum
+    request_checksum_required = (
+        operation_model.http_checksum_required
```

Comment:
Since httpChecksumRequired is being deprecated in favor of requestChecksumRequired, do we want to add something to make sure we will delete this once all operations migrate to requestChecksumRequired and remove the old logic?

Reply:
We will likely never be migrated in this case. If that does happen, that'll be a larger initiative.